Implement different meta-analysis modalities

rahularoradfs opened the pull request:
- Implemented three additional meta-analysis modalities, and re-added the fourth that Austin had previously implemented 
- Refactored code to enforce the type of `error` as `number []`, and coded all errors as asymmetric errors for consistency
- Fixed tooltips on bar charts to properly deal with asymmetric errors

atatmaja commented on src/metaAnalysis.ts:
```diff
@@ -1,30 +1,115 @@
 import { AirtableRecord, AggregationFactor, AggregatedRecord } from "./types"
+import { faPassport } from "@fortawesome/free-solid-svg-icons";
```
unused import

rahularoradfs replied:
removed

atatmaja commented on src/metaAnalysis.ts:
```diff
@@ -1,30 +1,115 @@
 import { AirtableRecord, AggregationFactor, AggregatedRecord } from "./types"
+import { faPassport } from "@fortawesome/free-solid-svg-icons";
+
+// BROKEN NOW: CONFIDENCE INTERVAL TOOLTIPS 
```
is this comment still accurate?

rahularoradfs replied:
no - removed

atatmaja commented on src/metaAnalysis.ts:
```diff
@@ -1,30 +1,115 @@
+            return Math.asin(Math.sqrt(p))
+        }
+        case "double_arcsin": {
+            let n = N * p
```
not changing the definition of n after its declaration, so we can just make it a const

rahularoradfs replied:
done

atatmaja commented on src/metaAnalysis.ts:
```diff
@@ -1,30 +1,115 @@
+        });
+
+        pooled_p = p_over_var_sum / inv_var_sum;
+        let conf_inter = [pooled_p - Z_SCORE * Math.sqrt(var_sum), pooled_p + Z_SCORE * Math.sqrt(var_sum)]
```
again we can make this a const

rahularoradfs replied:
done

atatmaja commented on src/metaAnalysis.ts:
```diff
@@ -1,30 +1,115 @@
+        error = [pooled_p - backTransform(conf_inter[0], method)!, backTransform(conf_inter[1], method)! - pooled_p];
     }
+
+    if (pooled_p - error[0] < 0) {error[0] = pooled_p}
```
nice typescript 1 line solution lol

atatmaja commented:
Maybe as a future step we can implement meta analysis as a class. That way we won't have to pass around state variables like the method type around a bunch of times. Also we could instantiate metaAnalysis objects with different method types in different parts of the app if we ever want to do something like that.

rahularoradfs commented:
refactored meta-analysis testing; fixed map tooltip for asymmetric error; added a different implementation of the double-arcsin transform and made that default
